docs/adr/0002-project-locks.md

```diff
@@ -0,0 +1,169 @@
+
+# ADR-0002: Project Locks
+
+- Date: 2023-05-09
+- Status: Draft 
+
+## Context and Problem Statement
+
+### Problem 
+
+There is a long-standing regression introduced by a PR to allow parallel plans to happen for projects within the same repository that also belongs in the same workspace. The error prompting users when attempting to plan is:
+```
```

Comment:
After commit 52883898979bf4ebc this is hard to trigger. In practice it requires a TryLock to be called while TryLockPull is held, and that only happens for a reasonably short time in buildAllProjectCommandsByPlan. 

TryLockPull can be killed and rewritten to actually lock the affected directories, then  this error will not be possible any longer.